Remotion/Data/DomainObjects/Queries/Configuration/QueryConfiguration.cs

```diff
@@ -63,6 +64,18 @@ public QueryConfiguration (params string[] configurationFiles) : this()
       }
     }
 
+    public QueryConfiguration (IAppContextProvider provider) : this (provider, new string[0])
+    {
+    }
+
+    public QueryConfiguration (params string[] configurationFiles) : this (new AppContextProvider(), configurationFiles)
+    {
+    }
+
+    public QueryConfiguration () : this (new AppContextProvider(), new string[0])
```

Comment:
Looks good. Just switch the order of the ctor methods, please: the empty ctor first, then the other overloads and finally the full implementation. (that's the typically used order with ctors and, I think, other overloads, too.)